src/main/java/io/harness/cf/client/api/Poller.java

```diff
@@ -61,20 +60,18 @@ protected void runOneIteration() {
             segments.stream()
                 .collect(Collectors.toMap(Segment::getIdentifier, segment -> segment)));
       }
-
+    } catch (Exception e) {
+      log.error("Failed to get FeatureConfig or Segments: {}", e.getMessage());
+    } finally {
       if (streamEnabled) {
         log.info("Switching to streaming mode.");
         cfClient.startSSE();
```

Comment:
Can this throw exception? we should put it in a function which does not throw.

Reply:
I dig a few level down into the underlining methods of startSSE(), they do not seem to throw exceptions.